server/sonar-server/src/main/java/org/sonar/server/permission/PermissionService.java

```diff
@@ -76,6 +77,11 @@ public PermissionService(DbClient dbClient, PermissionRepository permissionRepos
     return GlobalPermissions.ALL;
   }
 
+  public UserWithPermissionQueryResult findUsersWithPermission(PermissionQuery query) {
```

Comment:
Arg ! So why didn't you have called directly ComponentFinder ?
ComponentService must be removed, and permission must be checked in the WS